src/pages/AppList/FiltersAndSorts.ts

```diff
@@ -112,6 +140,39 @@ const filterByIstioSidecar = (items: AppListItem[], istioSidecar: boolean): AppL
   return items.filter(item => item.istioSidecar === istioSidecar);
 };
 
+const filterByIstioType = (items: AppListItem[], istioTypes: string[]): AppListItem[] => {
+  return items.filter(item => {
+    if (istioTypes.length > 0) {
+      if (istioTypes.includes(IstioTypes.virtualservice.name) && item.virtualServices.length > 0) {
+        return true;
+      }
+      if (istioTypes.includes(IstioTypes.destinationrule.name) && item.destinationRules.length > 0) {
+        return true;
+      }
+      if (istioTypes.includes(IstioTypes.gateway.name) && item.gateways.length > 0) {
+        return true;
+      }
+      if (istioTypes.includes(IstioTypes.authorizationpolicy.name) && item.authorizationPolicies.length > 0) {
+        return true;
+      }
+      if (istioTypes.includes(IstioTypes.peerauthentication.name) && item.peerAuthentications.length > 0) {
+        return true;
+      }
+      if (istioTypes.includes(IstioTypes.sidecar.name) && item.sidecars.length > 0) {
+        return true;
+      }
+      if (istioTypes.includes(IstioTypes.requestauthentication.name) && item.requestAuthentications.length > 0) {
+        return true;
+      }
+      if (istioTypes.includes(IstioTypes.envoyfilter.name) && item.envoyFilters.length > 0) {
+        return true;
+      }
+      return false;
+    }
+    return true;
+  });
+};
```

Comment:
In case you'd have a reference variable with all the objects, this match would be something similar to:

```typescript
return items.filter(item => item.references.filter(ref => istioTypes.includes(ref.kind)).length !== 0)
```